Event/FilterLocaleSwitchEvent.php

```diff
@@ -36,7 +36,7 @@ class FilterLocaleSwitchEvent extends Event
      */
     public function __construct(Request $request, $locale)
     {
-        if (!is_string($locale) || null == $locale || '' == $locale) {
+        if (!is_string($locale)) {
```

Comment:
Why is it important that empty strings are not allowed?
But ok, let's add ``|| !empty($locale)``.